src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  const ADD_PROPERTIES = 'addProperties';
+  const REMOVE_PROPERTIES = 'removeProperties';
+  const CLEAR = 'clear';
+
+  let tempObject = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ADD_PROPERTIES:
+        Object.assign(tempObject, action.extraData);
+        break;
+
+      case REMOVE_PROPERTIES:
+        for (const key of action.keysToRemove) {
+          delete tempObject[key];
+        }
+        break;
+
+      case CLEAR:
+        tempObject = {};
+        break;
+    }
```

Comment:
[Checklist](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) p. 3

Reply:
thx for reviewing,
it's allright now